# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1236624779 14400
#      Mon Mar 09 14:52:59 2009 -0400
# Node ID b8ed0262b11efd23f80c223b103db060fd4adb2a
# Parent  6994961bf1f41e43e91f72f2240c642863d1fc4c
lookfor.m: also match function names

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,12 @@
 2009-03-09  John W. Eaton  <jwe@octave.org>
 
+	* help/lookfor.m (search_cache): Also match function names.
+
 	* help/gen_doc_cache.m, help/lookfor.m: Use doc-cache instead of
 	DOC for doc cache file.
 
 2009-03-09  Jaroslav Hajek  <highegg@gmail.com>
 
 	* general/accumarray.m: Reorder tests. Call either "sparse" or
 	__accumarray_sum__ for the default summation case.
 	* statistics/base/histc.m: Reimplement using lookup & accumarray.
diff --git a/scripts/help/lookfor.m b/scripts/help/lookfor.m
--- a/scripts/help/lookfor.m
+++ b/scripts/help/lookfor.m
@@ -168,18 +168,18 @@ function [out_fun, out_help_text] = look
     ## Return the results instead of displaying them
     out_fun = fun;
     out_help_text = help_text;
   endif
 endfunction
 
 function [funs, help_texts] = search_cache (str, cache_file, search_type)
   load (cache_file);
-  if (! isempty(cache))
-    tmp = strfind (cache (search_type, :), str);
-    cache_idx = find (!cellfun ("isempty", tmp));
+  if (! isempty (cache))
+    t1 = strfind (cache (1, :), str);
+    t2 = strfind (cache (search_type, :), str);
+    cache_idx = find (! (cellfun ("isempty", t1) & cellfun ("isempty", t2)));
     funs = cache (1, cache_idx);
     help_texts = cache (3, cache_idx);
   else
     funs = help_texts = {};
   endif
 endfunction
-
